### DISCUSSION TITLE
What is required to go live?

### DISCUSSION THREAD

**Discussion by leamas** [opencpn-manuals/main#16] [General]
As I see it, the basic pieces are in place for the development and the user manual.  OTOH, there is a need for all sorts of polish, a work which will be ongoing.

Question is then: what is required for us to announce the new manual(s) as available? As I see it, we should have a list of issues we should fix. A list we agree on that once those issues are fixed, we announce the project. Not as "ready", just won't happen, but "available".

@Rasbats:  @rgleason : Thoughts?

**Comment by rgleason** [comment]
You can go live now as far as I can see. It will hopefully get better. Are most links fixed?

Rasbats when you guys decide can you put the new link and announcement in the wiki, but do not delelte any of the wiki!!
Do not delete any of the wiki please!!!

**Comment by Rasbats** [comment]
I would like to see all the PR made to the dev repos first. Then the devs have a page in the manual they can learn from/edit/play.
With this done we can remove the opencpn-manuals fork plugin stuff for edit_url and use the dev repo.

Edit: Try to make more sense.

**Comment by leamas** [comment]
> I would like to see all the PR made to the dev repos first

Agreed.  We can not force anyone to take a PR, but we can do our part which is to file a PR for each and every forked plugin. We need to file an issue on this.

**Comment by leamas** [comment]
Filed https://github.com/opencpn-manuals/plugins/issues/23 about making a PR.

**Reply by leamas** [reply]
Seems that  https://github.com/opencpn-manuals/plugins/issues/23 is at a point where we have file a PR for all forked plugins. Now, all depends on if/when these are accepted. In other words, it is not a blocking issue.

Is there anything else we should do? Somehow, I'm thinking of a review, at least to a point where the have been four eyes, not just two, on central pieces. However, reviewing is hard work, and we might lack man-hours  or motivation enough.

Thoughts?

**Comment by rgleason** [comment]
You mean read each manual in html form and make a summary and fix the worst?

If so, what about make a list here and a link to the   io and we just put our name when read. The summary should be filed under the plugins issues for author to fix.

**Reply by leamas** [reply]
> You mean read each manual in html form and [...]

yes 

> You mean read each manual in html form and make a summary  and  fix the worst?

I could create a review PR. In such a PR one can make notes directly attached to the sources; it might be a better alternative. Still when you make a review you don't "fix" things, you just make remarks. Changes are done by the reviewee. 

Any way, if we should do it we need to assign different parts between us. And we need to agree to do it. This is not to be taken lightly...

**Comment by rgleason** [comment]
I am in process on all these. Listed https://github.com/opencpn-manuals/plugins/issues/25#issuecomment-909209200

I will try to do an initial review of all first, then begin at the top and fix links and images, and updates.  This will be reported in the link above.

**Comment by rgleason** [comment]
I have reviewed and updated all of the PI I am responsible for and have pushed those changes up to my remote repositories.

Also this morning I wrote Mauro Calvi a short email asking him to merge the PRs for the manual.

**Comment by rgleason** [comment]
I saw Dashboard Tactics is removed.
For Dashboard Tactics I will make a single page that refers to Petri's documentation.
and will advise when it is done.

**Comment by rgleason** [comment]
Tonight I may be able to use opencpn-manuals/plugins.

**Comment by leamas** [comment]
@rgleason @Rasbats : Are we about to reach the point that we think the Plugins Manual is good enough to be announced?

**Reply by leamas** [reply]
@rgleason @Rasbats : ^   A week later: what is missing?

**Comment by rgleason** [comment]
I made the page last night.  dashboard_tactics_pi   manual is https://github.com/rgleason/dashboard_tactics_pi/tree/master/manual and it works.

I also locally changed the nav.adocs and tried to update site.yml, source-state.sh, sources.state file and sources folder, but it failed.

We need a "How to Add a Plugin to the Plugin Manual"!

I don't understand what is happening between site.yml , source-state.sh and sources and when and under what circumstances to use source-state.sh (update restore and save).

Should the sources.state file list use the plugins SHA for the most recent commit or is it something else??
Because it kept failing.  I entered the SHA for my PI   commit cd20bb853f558eadcb098d4e8c0bc6d18f419c1a   and it failed.

**Reply by leamas** [reply]
The "right" way to use sources and source-state.sh is like this (omitting all shortcuts):
  - Checkout a version of sources.state which is known to work.
  - Remove all directories under sources (but keep .gitkeep)
  - Restore the sources: `./source-state.sh restore`
  - If you want to update for example weather-routing_pi:

         $ ./source-state.sh update weather-routing_pi
  - Make a local build and check the results: `antora site.yml`
  - Record the current state and commit it
  
        $ ./source-state.sh save
        $ git add sources.state
        $ git commit -m "sources: Update weather-routing_pi."

And that's it... This is an ordered way to get the correct info into sources.state. It takes some time, but works.

**Reply by rgleason** [reply]
Alec, that is very helpful, and I am going to keep that in my cheat sheet, but  I don't think that answers what needs to be done to get the manual for dashboard_tactics included.

> I made the page last night. dashboard_tactics_pi manual is https://github.com/rgleason/dashboard_tactics_pi/tree/master/manual and it works.

How do I add that manual?

**Comment by rgleason** [comment]
As far as I am concerned you can announce now.

**Comment by Rasbats** [comment]
Just corrected the commit for oesenc in sources.state. Now working.
Looks good from my end. 
When time allows I will take another look at Beta/Main. The use of partials (nav-plugins.adoc/plugins.adoc was to allow navigation between the two modules - master and beta. Perhaps this can be simplified.

**Comment by Rasbats** [comment]
We need to keep opencpn-manuals/main in sync with opencpn-manuals/plugins. Opening the main oesence link does not work. I could correct sources.state in main but ...

**Reply by leamas** [reply]
My plan is to announce opencpn-manuals/main  as the "official" manual. This way, we can experiment in opencpn-manuals/plugins without risking the main manual. But  opencpn-manuals/main must be updated, agreed.

**Comment by rgleason** [comment]
If that is the case, I would like to get Dashboard_tactics into the main manual too.

**Comment by leamas** [comment]
@rasbats @rgleason:

We have a preliminary version of the complete manual available at https://opencpn-manuals.github.io/main. 

Please take a look at the bits you care about, not least in the development part which hasn't got much love lately.  If it's ok with you, we make a final build tomorrow and announces it.

**Comment by Rasbats** [comment]
Publish please.

**Comment by Rasbats** [comment]
@rgleason : Howto for adding a plugin to the manual will appear when time allows. I think it should probably go in development somewhere.

**Reply by rgleason** [reply]
Good!

**Reply by rgleason** [reply]
I think you should also include a link to the Discussion solution  https://github.com/opencpn-manuals/main/discussions/20#discussioncomment-1363128   so that the discussion with Alec can be read as it is instructive and show some of the pitfalls.

NOTE: I still don't know what was causing all those FATALS although Alec thinks it is from the comments // that he found in nav.adoc and index.adoc that I did not do.  I think it is something else that is causing that on my machine.

**Reply by rgleason** [reply]
We also need a good generic site.yml example and antora.yml example.

**Reply by rgleason** [reply]
What about just having it in your Plugins  "Authoring" section?

1. How to Add a Plugin to the Manual
2. How to author with Asciidoc/Antora - Windows setup - pandoc conversion, antora, vscode, firefox extension, etc.
4. How to author with Asciidoc/Antora - Linux - 
5. Site.yml  standard
6. Antora.yml standard
5. other?

**Comment by rgleason** [comment]
There is no way I am going to be able to do justice to the TestPlugin Template Dev Docs. That is something Jon and I have to work on and I have not had much time to do that.  Also I have many plugins that really need to be updated. So just go with what we have there now. It will have to be reworked completely anyway, later.  I will try to get Jon involved, but it might help if you invited him to the opencpn-manuals so he can work on it. Thanks.   

Other than that I think it looks pretty good and I like the clearer menu at the beginning of this link.  
There are are few little index issues, like  dashboard_tactics is both a Log and for Sailing and somehow that got changed but that does not really matter.  It is listed under Logs on the opencpn.org website, but it really should be under sailing I think.

Publish!!  You guys have done a remarkable job, with a difficult system that is a bit finicky IMHO. (I like to just hit SAVE, it is so simple!)
Congratulations on your success.

I am really looking forward to seeing how the new "asciidoc/antora  distributed plugin manuals"   help us all integrate the plugin documentation with Alec's wonderful "tarballs"  and have it accessible from within the plugin or Opencpn!

**Comment by rgleason** [comment]
Mike, on a separate topic.  Could you sometime take a look at 
https://github.com/opencpn-manuals/main/discussions/22

and perhaps simplify and try to eliminate duplication of index.adoc and nav.adocs in the menus?
Perhaps what I have suggested is not possible because you want to have Beta and Main plugins in separate expandible tabs in the index, but there should be a way to make this simpler.

**Comment by Rasbats** [comment]
https://github.com/opencpn-manuals/plugins/issues/72#issuecomment-924698844
